lib/spack/spack/main.py

```diff
@@ -363,7 +363,15 @@ def make_argument_parser(**kwargs):
         metavar='DIR', help="add a custom configuration scope")
     parser.add_argument(
         '-d', '--debug', action='store_true',
-        help="write out debug logs during compile")
+        help="write out basic debug messages")
+    parser.add_argument(
```

Comment:
If we want to keep the existing behavior, `--debug` should be equivalent to `-dd`.

Reply:
The way `argparse` works, `--debug` will just be `-d` and `--debug --debug` will be `-d`.  Are you suggesting that level 1 should be more verbose?  I'm actually fine with that.

Oh I see this isn't counting -- see my comment on counting below.

I think we should count and make the help message here should be more descriptive, e.g.:

```
write out debug messages (more d's for more verbosity: -d, -dd, -ddd, etc.)
```

Reply:
According to my notes, existing debug messages were supposed to be at level 2 (or `-dd`) by default and a bunch of the current build-related `tty.msg` output should be output as part of the less verbose, or basic, debug output of `-d` (or `--debug`).

We've gotten complaints about the amount of debugging output so I thought this was a good move (i.e., push all debug messages to a more verbose "level").

I will change the code so the default debug level is 1 to preserve the existing debug output setting behavior.

Reply:
As for counting, I will l look into that change as well.

Reply:
Made both changes.